meta-mbl-distro/recipes-security/ecryptfs-utils/files/ecryptfs-init.sh

```diff
@@ -0,0 +1,250 @@
+#!/bin/bash
+#
+###############################################################################
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# ecryptfs homepage: http://ecryptfs.org/
+# Summary: Utilities for the Linux cryptographic filesystem ecryptfs.
+#
+# Description
+# ===========
+#  This script is used to mount an ecryptfs ciphered directory at system
+#  startup. If the ciphered directory is being mounted for the first time
+#  (e.g. when the system boots after flashing an image) then the required
+#  configuration is generated and stored. The configuration includes:
+#  - An ecyptfs FEKEK (File Encryption Key Encryption Key).
+#  - A passphrase to protect the FEKEK, and the storing of the passphrase
+#    in a file.
+#  - An entry in /etc/fstab specifying the mount operation. This uses the
+#    passphrase file.
+#  - The creation of the upper (unencrypted) and lower (encrypted) storage
+#    directories.
+#
+#  The following operations happen during boot:
+#  - The passphrase protected FEKEK is added to the user kernel keyring.
+#  - The unecrypted (upper) directory is mounted on the encrypted (lower)
+#    directory by ecryptfs to provide deciphered access to store files.
+#    Mount options in fstab specify the FEKEK key signature so ecryptfs
+#    can read the cipher key from the keyring. The passphrase is also read
+#    to decode the FEKEK.
+#
+#  The upper and lower directory terminology is explained in:
+#    <src/>/Documentation/filesystems/overlayfs.txt
+#
+# Outstanding Issues
+# ==================
+#  - The passphrase should not be stored on the filesystem, but retrieved
```

Comment:
I am checking how we can make it working for Kepco. For know Kepco's project does not have additional storage for passphrase storing

Reply:
Ultimately, some ecryptfs private key needs to be stored on the system, and its likely to be target specific secure storage e.g. the CAAM on nxp warp7/imx8. I created this issue for specifying how:

https://jira.arm.com/browse/IOTMBL-2342

Jeremy has taken this issue and is working on it in the current sprint. As the answer is likely to be platform specific, you'lll need your own answer for the kepco project, but this can be informed by iotmbl-2342.